frontend/src/hooks/useApi.ts

```diff
@@ -48,32 +46,35 @@ export interface PimsAPI extends AxiosInstance {
 
 export const useApi = (): PimsAPI => {
   const dispatch = useDispatch();
-  const axios = useMemo(() => CustomAxios() as PimsAPI, []);
 
-  axios.interceptors.request.use(
-    config => {
-      config.headers.Authorization = `Bearer ${store.getState().jwt}`;
-      dispatch(showLoading());
-      return config;
-    },
-    error => {
-      dispatch(hideLoading());
-      return Promise.reject(error);
-    },
-  );
-
-  axios.interceptors.response.use(
-    config => {
-      dispatch(hideLoading());
-      return config;
-    },
-    error => {
-      dispatch(hideLoading());
-      return Promise.reject(error);
-    },
-  );
-
-  axios.isPidAvailable = useCallback(
+  const getAxios = useCallback(() => {
```

Comment:
Alright, going to go through some of my decision making on this:
1) Original case, axios instance is recreated every time as it was not memoized. This causes any memoizing dependent on functions in this hook to recreate every render.
2) Ideal case, have one instance of axios, re-use that instance for all requests. All of the functions added to the axios object should remain the same as they are wrapped in a callback. **This caused major performance issues, also jwt token was not renewed correctly**
3) This Implementation. What I've ended up going with here is a getter that is invoked whenever an instance of axios is required. This is also more in line with our other api hooks that call customAxios() directly. The downside of this is that the axios instance is being rebuilt for each request. However, based on testing this method is much more performant. Also updated the returned object to just include the function. We have no use case for returning the actual axios instance here.